### ISSUE TITLE
Something eats up a lot of time

### ISSUE THREAD

**Issue by RSDuck** [pragmagic/vscode-nim#44]
It looks like that something(probably the RPC library) eats up a lot of time. At first I thought the problem is related to nimsuggest but it isn't(https://github.com/nim-lang/nimsuggest/issues/55)

In all tests I used the following file:
y.nim
```nimrod
import opengl



```

I added the following code to measure the performance within the extension:
```nimrod
let startTime = Date.now();
let ret = await desc.rpc.callMethod(new elparser.ast.SExpSymbol(NimSuggestType[suggestType]), normalizedFilename, line, column, dirtyFile);
let endTime = Date.now();
console.log("Request time: " + (endTime - startTime))
```
I moved the cursor into the third line of y.nim and it took it outputted a request time of around 7 seconds.

The I tried the following test program(it needs nimsuggest to be started with the arguments --port:2811 y.nim):
```nimrod
import times, net

var 
    line = ""
    totalTime = 0'f32
    measured = 0
for i in 0..2:
    var socket = newSocket()
    socket.connect("127.0.0.1", Port(2811))
    let startTime = cpuTime()
    socket.send("sug y.nim:3:0\n")
    var responseLines = 0
    while true:
        socket.readLine(line, 1000)
        if line == "":
            break
        inc(responseLines)
    let endTime = cpuTime()

    echo "Duration: ", (endTime - startTime) * 1000'f32, "ms"
    echo "Got ", responseLines, " results"
    totalTime += (endTime - startTime)
    inc(measured)

echo "Average time: ", totalTime / measured.toFloat * 1000'f32, "ms"
```
And it outputted always something like this:
```
Duration: 496.0ms
Got 8708 results
Duration: 427.0000000000001ms
Got 8708 results
Duration: 236.9999999999999ms
Got 8708 results
Average time: 386.6666555404663ms
```

So as you can see it isn't nimsuggests fault. I'm thinking the problem is related to elrpc but I'm not sure about this.

**Comment by RSDuck** [comment]
I've put some further research into it and created a version of vscode-nim which uses the old socket based protocol instead of epc and it's actually much faster(the code thrown together, mostly from old an old version of vscode-nim which used this protocol):
https://github.com/RSDuck/vscode-nim/tree/oldsockets

![gif4](https://cloud.githubusercontent.com/assets/9352526/23910859/51d93cbc-08db-11e7-9bff-5943df1e1b4a.gif)
vs:
![gif5](https://cloud.githubusercontent.com/assets/9352526/23913463/64d75b34-08e3-11e7-9c8c-c695e432fb4d.gif)

**Comment by kosz78** [comment]
The better option will be improve EPC because this protocol returns lenght of message that guarantee that we read full message

**Comment by RSDuck** [comment]
already fixed with #48, I forgot to close it
